src/store/modules/AccountsModule.js

```diff
@@ -41,15 +49,56 @@ class AccountsModule extends VuexModule {
     return !!this.settings.otpEnabled;
   }
 
+  get ethBalance() {
+    if (!this.isBalanceLoading && this.balance) {
+      return fromWei(this.balance);
+    }
+    return '0';
+  }
+
+  /**
+   * @param {object} newSettings
+   */
+  @Action
+  async changeSettings(newSettings) {
+    const oldSettings = this.settings;
+    this.settings = newSettings;
+
+    const isNetworkChanged = oldSettings.net !== newSettings.net;
+    if (isNetworkChanged) {
+      setWeb3Network(newSettings.net);
+    }
+
+    this.resubscribeBalance({ isNetworkChanged, newSettings, oldSettings });
+  }
+
+  @Action
+  async resubscribeBalance({ isNetworkChanged, newSettings, oldSettings }) {
+    if (!this.isBalanceAutoUpdate) {
+      return;
+    }
+
+    const isLastAccountChanged =
+      newSettings.lastActiveAccount &&
+      oldSettings.lastActiveAccount &&
+      oldSettings.lastActiveAccount !== newSettings.lastActiveAccount;
+
+    const isBalanceUpdateStopped = !this.isBalanceLoading && !this.balance;
```

Comment:
So, what about cases when we really have zero balance? Is it really sign of balance updates stopping?